printing_test.go

```diff
@@ -0,0 +1,120 @@
+package terminal
+
+import (
+	_ "embed"
+	"testing"
+	"unicode/utf16"
+
+	"fyne.io/fyne/v2"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHandleOutput_PrintMode(t *testing.T) {
+	tests := map[string]struct {
+		inputSeq string
+	}{
+		"pdf printing": {
+			inputSeq: esc("_set printer:_editor:tmp/DBRRPT-20231026-022112-27.3446959766687.pdf") + "\000" + esc("\\"),
+		},
+	}
+
+	// Iterate through the test cases
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			terminal := New()
+			terminal.handleOutput([]byte(test.inputSeq))
+		})
+	}
+}
+
+func TestHandleOutput_Printing(t *testing.T) {
+	tests := map[string]struct {
+		inputSeq              []byte
+		expectedPrintingState bool
+		expectedPrintData     []byte
+		expectedSpooledData   []byte
+		expectedScreenData    string
+	}{
+		"start printing": {
+			inputSeq:              []byte(esc("[5ithisshouldbeprinted")),
+			expectedPrintData:     []byte("thisshouldbeprinted"),
+			expectedPrintingState: true,
+		},
+		"complete printing": {
+			inputSeq:              []byte(esc("[5i") + "thisshouldbeprinted" + esc("[4i")),
+			expectedSpooledData:   []byte("thisshouldbeprinted"),
+			expectedPrintingState: false,
+		},
+		"printing with embedded esc": {
+			inputSeq:              []byte(esc("[5i") + esc("����B�") + esc("[4i")),
+			expectedSpooledData:   []byte(esc("����B�")),
+			expectedPrintingState: false,
+		},
+		"UTF-8 Content": {
+			inputSeq:              []byte(esc("[5i") + "Hello, 世界!" + esc("[4i")),
+			expectedSpooledData:   []byte("Hello, 世界!"),
+			expectedPrintingState: false,
+		},
+		"UTF-16 Content": {
+			inputSeq: []byte(esc("[5i") + string(utf16.Decode([]uint16{
+				0x0048, 0x0065, 0x006c, 0x006c, 0x006f, 0x002c, 0x0020, 0x4e16, 0x754c, 0x0021,
+			})) + esc("[4i")),
+			expectedSpooledData:   []byte("Hello, 世界!"),
+			expectedPrintingState: false,
+		},
+		"ISO-8859-1 Content": {
+			inputSeq:              []byte(esc("[5iH") + "\xe9llo, W\xf6rld!" + esc("[4i")),
+			expectedSpooledData:   []byte{0x48, 0xe9, 0x6c, 0x6c, 0x6f, 0x2c, 0x20, 0x57, 0xf6, 0x72, 0x6c, 0x64, 0x21},
+			expectedPrintingState: false,
+		},
+		"when print is mid buffer": {
+			inputSeq:            []byte("abc" + esc("[5i") + "def" + esc("[4i") + "ghi"),
+			expectedScreenData:  "abcghi",
+			expectedSpooledData: []byte("def"),
+		},
+	}
+
+	// Iterate through the test cases
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			terminal := New()
+			terminal.Resize(fyne.NewSize(50, 50))
+			var spooledData []byte
+			terminal.printer = PrinterFunc(func(d []byte) {
+				spooledData = d
+			})
+			terminal.handleOutput(test.inputSeq)
+
+			assert.Equal(t, test.expectedScreenData, terminal.content.Text())
+			assert.Equal(t, test.expectedPrintingState, terminal.state.printing)
+			assert.Equal(t, test.expectedSpooledData, spooledData)
+			assert.Equal(t, test.expectedPrintData, terminal.printData)
+		})
+	}
+}
+
+//go:embed test_data/chn.pdf
```

Comment:
The content of this file makes me nervous - can we have a file that our developers can read so can make sure it's suitable...

Reply:
can you please make a suggestion?

Reply:
oh sorry are you talking about the actual file itself? rather than the naming of the go file. printing_test.go?
theres a reason for it being in this language, its to ensure the byte for byte data for the extended set is preserved. This file was the reason I had to change the main loop in handleOutput.

Reply:
I was meaning the content of the (Chinese?) text.
Also what is its license etc...

Reply:
The source for this file https://www.ohchr.org/sites/default/files/UDHR/Documents/UDHR_Translations/chn.pdf
seamed a safe source for a real pdf

Reply:
You could substitute a binary file of your own choosing.  For example, make a PDF of a page of your source code.  The licensing would be defined: it's your file and the license can be the same as the programs.